Source/Core/Core/IOS/USB/Emulated/Skylander.cpp

```diff
@@ -0,0 +1,751 @@
+{
+  return true;
+}
+
+int SkylanderUsb::CancelTransfer(u8 endpoint)
+{
+  return IPC_SUCCESS;
+}
+
+int SkylanderUsb::ChangeInterface(const u8 interface)
+{
+  DEBUG_LOG_FMT(IOS_USB, "[{:04x}:{:04x} {}] Changing interface to {}", m_vid, m_pid,
+                m_active_interface, interface);
+  m_active_interface = interface;
+  return 0;
+}
+
+int SkylanderUsb::GetNumberOfAltSettings(u8 interface)
+{
+  return 0;
+}
+
+int SkylanderUsb::SetAltSetting(u8 alt_setting)
+{
+  return 0;
+}
+
+int SkylanderUsb::SubmitTransfer(std::unique_ptr<CtrlMessage> cmd)
+{
+  DEBUG_LOG_FMT(IOS_USB,
+                "[{:04x}:{:04x} {}] Control: bRequestType={:02x} bRequest={} wValue={:04x}"
+                " wIndex={:04x} wLength={:04x}",
+                m_vid, m_pid, m_active_interface, cmd->request_type, cmd->request, cmd->value,
+                cmd->index, cmd->length);
+
+  cmd->expected_time = Common::Timer::NowUs() + 100;
+  auto& system = Core::System::GetInstance();
+  auto& memory = system.GetMemory();
+  u8* buf = memory.GetPointerForRange(cmd->data_address, cmd->length);
+  std::array<u8, 64> q_result = {};
+  std::array<u8, 64> q_data = {};
+  // Control transfers are instantaneous
+  switch (cmd->request_type)
+  {
+  // HID host to device type
+  case 0x21:
+    switch (cmd->request)
+    {
+    case 0x09:
+      switch (buf[0])
+      {
+      case 'A':
+      {
+        // Activation
+        // Command	{ 'A', (00 | 01), 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00,
+        // 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00 }
+        // Response	{ 'A', (00 | 01),
+        // ff, 77, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00, 00,
+        // 00, 00, 00, 00, 00, 00, 00, 00 }
+        // The 3rd byte of the command is whether to activate (0x01) or deactivate (0x00) the
+        // portal. The response echos back the activation byte as the 2nd byte of the response. The
+        // 3rd and 4th bytes of the response appear to vary from wired to wireless. On wired
+        // portals, the bytes appear to always be ff 77. On wireless portals, during activation the
+        // 3rd byte appears to count down from ff (possibly a battery power indication) and during
+        // deactivation ed and eb responses have been observed. The 4th byte appears to always be 00
+        // for wireless portals.
+
+        // Wii U Wireless: 41 01 f4 00 41 00 ed 00 41 01 f4 00 41 00 eb 00 41 01 f3 00 41 00 ed 00
+        if (cmd->length == 2 || cmd->length == 32)
+        {
+          q_data = {buf[0], buf[1]};
+          q_result = {0x41, buf[1], 0xFF, 0x77, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                      0x00, 0x00,   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                      0x00, 0x00,   0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+          q_queries.push(q_result);
+          cmd->expected_count = 10;
+          g_skyportal.Activate();
```

Comment:
Why did you switch back to only using activate?

Reply:
Using the Deactivate method constantly causes the game to no longer recognise the figures on the portal, and the real trap team portal either didn't send deactivate commands or didn't respond to them, which is why I removed it.

Reply:
This sounds like there is something wrong in your code. As the real portal definitely changes behaviour on deactivate. Either way, it’s fine for emulation. It was mostly a power saving mode either way